dataset/except.cpp

```diff
@@ -10,8 +10,11 @@ namespace scipp::dataset::expect {
 void coordsAreSuperset(const DataArrayConstView &a,
                        const DataArrayConstView &b) {
   for (const auto &[dim, coord] : b.coords())
-    if (a.coords()[dim] != coord)
-      throw except::CoordMismatchError("Expected coords to match.");
+    if (a.coords()[dim] != coord) {
+      const std::string msg = "Mismatched coordinates: \"" + to_string(dim) +
+                              "\"\n" + to_string(a.coords()[dim]) +
+                              to_string(b.coords()[dim]);
```

Comment:
I think this defeats to purpose of adding the new exception constructor, since you end up not using it here.